File: src/main/java/com/jss/osiris/modules/osiris/tiers/service/AgeRangeService.java
package com.jss.osiris.modules.osiris.tiers.service;

import java.util.List;

import com.jss.osiris.modules.osiris.tiers.model.AgeRange;

public interface AgeRangeService {
    public List<AgeRange> getAgeRanges();

    public AgeRange getAgeRange(Integer id);

    public AgeRange addOrUpdateAgeRange(AgeRange ageRange);
}

File: client/src/app/modules/tiers/services/age.range.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { AgeRange } from '../../tiers/model/AgeRange';

@Injectable({
  providedIn: 'root'
})
export class AgeRangeService extends AppRestService<AgeRange>{

  constructor(http: HttpClient) {
    super(http, "tiers");
  }

  getAgeRanges() {
    return this.getList(new HttpParams(), "age-ranges");
  }
  
   addOrUpdateAgeRange(ageRange: AgeRange) {
    return this.addOrUpdate(new HttpParams(), "age-range", ageRange, "Enregistré", "Erreur lors de l'enregistrement");
  }

}
